### PR TITLE
enhancement #789: added the ability to get selective discipline for t…

### LINKED ISSUE #789: Add the ability to return student selective disciplines to bot
Your task is to improve the endpoints for receiving events in bot. You must add userId as query. And in this case, return selective disciplines of only this user with other events. If there is no such query, then return all selective disciplines with other events.

image
image

### PR DESCRIPTION
**Pull request by f-ogoog**
…elegram

- added the ability to get selective discipline for telegram
- changed 3 endpoints for telegram

closes #789

### CODE REVIEW

File: src/v2/api/services/ScheduleService.ts [hunk 1]
```diff
@@ -659,11 +663,27 @@ export class ScheduleService {
...
     week = week || await this.dateService.getCurrentWeek();
     const [firstWeek, secondWeek] = week % 2 === 0 ? [week-1, week] : [week, week+1];
-    const { events: firstWeekEvents } = await this.getGeneralGroupEvents(groupId, firstWeek);
-    const { events: secondWeekEvents } = await this.getGeneralGroupEvents(groupId, secondWeek);
+    const firstWeekEvents = await this.getGroupEventsForTelegram(groupId, firstWeek, userId);
+    const secondWeekEvents = await this.getGroupEventsForTelegram(groupId, secondWeek, userId);
     return { firstWeekEvents, secondWeekEvents };
   }
+
+  async getGroupEventsForTelegram (groupId: string, week: number, userId) {
```
Comment (SharpDevOps10) [inline src/v2/api/services/ScheduleService.ts:674]: Add type for `userId`

**Review by SharpDevOps10 — CHANGES_REQUESTED** [review]

File: src/v2/api/responses/DisciplineTypeResponse.ts [hunk 2]
```diff
@@ -2,12 +2,17 @@ import { ApiProperty } from '@nestjs/swagger';
...
 
 export class DisciplineTypeResponse {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Discipline type id',
+  })
     id: string;
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Discipline`s id',
```
Comment (SharpDevOps10) [inline src/v2/api/responses/DisciplineTypeResponse.ts:10]: Rename to: `Discipline\'s id`

File: src/v2/api/responses/MainEventInfoResponse.ts [hunk 3]
```diff
@@ -2,20 +2,29 @@ import { ApiProperty } from '@nestjs/swagger';
...
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Event`s id',
+  })
     id: string;
 
-  @ApiProperty()
+  @ApiProperty({
+    description: 'Event`s name',
+  })
```
Comment (SharpDevOps10) [inline src/v2/api/responses/MainEventInfoResponse.ts:12]: * Rename to: `'Event\'s id'`
* Rename to: `'Event\'s name'`

File: src/v2/api/responses/TelegramGeneralEventInfoResponse.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { ApiProperty } from '@nestjs/swagger';
-import { GeneralEventInfoResponse } from './MainEventInfoResponse';
-
-export class TelegramGeneralEventInfoResponse extends GeneralEventInfoResponse {
-  @ApiProperty()
+import { MainEventInfoResponse } from './MainEventInfoResponse';
+export class TelegramEventInfoResponse extends MainEventInfoResponse {
+  @ApiProperty({
+    description: 'Event`s url',
```
Comment (SharpDevOps10) [inline src/v2/api/responses/TelegramGeneralEventInfoResponse.ts:5]: Rename to: `Event\'s url`

**Review by SharpDevOps10 — APPROVED** [review]

**Review by JessFreak — CHANGES_REQUESTED** [review]
endpoint returns incorrect selective disciplines

**Review by fokaaas — APPROVED** [review]

**Pull request merged**